caikit_nlp/toolkit/text_generation/tgis_utils.py

```diff
@@ -37,9 +37,19 @@
 
 GENERATE_FUNCTION_TGIS_ARGS = """
     {}
-    preserve_input_text: str
+    preserve_input_text: bool
         Whether or not the source string should be contained in the generated output,
         e.g., as a prefix.
+    input_tokens: bool
+        Whether or not to include list of input tokens.  
+    generated_tokens: bool
+        Whether or not to include list of individual generated tokens.  
+    token_logprobs: bool
+        Whether or not to include logprob for each returned token.  
+        Applicable only if generated_tokens == true and/or input_tokens == true
+    token_ranks: bool
+        Whether or not to include rank of each returned token.     
```

Comment:
`token_ranks` is added but there is no output allowing for `rank` information to be exposed. Is this intentional?

Reply:
yes, I had some trouble initially with figuring out the output for the `token_ranks` parameter.
Since it wasnt to core ask, I exposed the parameter but have not adjusted the output for it in the data-model or guardrails